### ISSUE TITLE
In a large application in dev mode the error ERR_CONNECTION_REFUSED 

### ISSUE THREAD

**Issue by Flashantik** [nuxt/cli#335]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v18.19.0
- Nuxt Version:     3.10.0
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      srcDir, devtools
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://github.com/Flashantik/nuxt-3.10.-slow-dev-mode

### Describe the bug

We have a large application, however after updating nuxt@3.7 dev mode started to slow down a lot and often cannot render all components 
I was able to recreate this behavior in a clean project with a lot of simple components...

To get the error all you need to do is run npm run dev and go to localhost:3000. 
Many components in the network will be in pending mode and after some time 500 error or ERR_CONNECTION_REFUSED error will be displayed. If there is no error in the console it is enough to reload the page.

build mode works fine. If you install nuxt@3.6.5 there are no such problems with dev mode. 
<img width="1191" alt="Screenshot 2024-02-05 at 21 49 46" src="https://github.com/nuxt/nuxt/assets/25429994/863e6592-0a4b-4c58-998a-1fae65c06690">

### Logs

```shell-script
Client logs: 

vue-router.mjs:3471 TypeError: Failed to fetch dynamically imported module: http://localhost:3000/_nuxt/pages/index.vue
triggerError @ vue-router.mjs:3471
(anonymous) @ vue-router.mjs:3184
nuxt.js:98 [nuxt] error caught during app initial
[Vue Router warn]: uncaught error during route navigation:

[nuxt] error caught during app initialization TypeError: Failed to fetch dynamically imported module

Failed to load resource: net::ERR_CONNECTION_REFUSED
Failed to load resource: net::ERR_SOCKET_NOT_CONNECTED
Hydration class mismatch

Server Logs (sometimes) 

ERROR  [unhandledRejection] connect ETIMEDOUT 127.0.0.1:61574                                                   9:49:52 PM

  at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1595:16)
```

**Comment by ThomasBerneHCSE** [comment]
Same problem occurs on nuxt 3.11.1 with nitropack 2.9.X

**Comment by Flashantik** [comment]
> Same problem occurs on nuxt 3.11.1 with nitropack 2.8.X

Yes, but it's hardly a nitropack problem. nuxt@3.6.5  works fine with nitropack@2.8.1, but nuxt@3.7.0 with nitropack@2.8.1 is already slow and buggy. This continues on nuxt@3.11.1.

**Comment by joaomarcelofm** [comment]
This is happening in our project that isn't that big even (±400 components). It does indeed seem to be mainly a Nuxt issue, as @Flashantik pointed out in https://github.com/nuxt/nuxt/issues/23832.

**Comment by Flashantik** [comment]
> This is happening in our project that isn't that big even (±400 components). It does indeed seem to be mainly a Nuxt issue, as @Flashantik pointed out in [nuxt/nuxt#23832](https://github.com/nuxt/nuxt/issues/23832).

It's not just about the components, but also about composables, utils pages, and so on. Each file in dev mode is loaded separately on the network, I think this is the problem..

By the way, as a solution to the problem, I found using bun instead of node, it is capable of consistently delivering about 10,000 files

**Comment by robinscholz** [comment]
Facing the same problem!

**Comment by flakerimi** [comment]
Same here,loses connection with reka ui, but after some times it works again untill I refactor imports.
Like when I move / rename a component it happens. Still not sure how to reproduce or fix.

**Comment by Flashantik** [comment]
> I was not able to reproduce with your original repo

Hi! I’ve updated Nuxt to the latest version (nuxt @ 3.17.6) and things are definitely better now. However, the error can still appear if you refresh the page several times. Since the original issue, I’ve also upgraded to Node v22.16.0 and moved to a much more powerful device, which might have affected things. If you need any extra details or reproduction steps, just let me know—at this point the way it works is already quite satisfactory.

https://github.com/user-attachments/assets/307e7df0-d926-4a65-b088-fca7a780c346

**Comment by flakerimi** [comment]
I dont think it has to do with nuxt it self, but yes latest version is
better!

Miniera e Kreativitetit

**Comment by hdsandbox** [comment]
I was able to reproduce the error with https://github.com/Flashantik/nuxt-3.10.-slow-dev-mode. We are experiencing the same problem with our main frontend project.

Operating System: macOS Sequoia
Browser: Google Chrome 138.0.7204.184 (Official Build) (arm64)
Nuxt 3.18.1 with Nitro 2.12.4
Node: v20.19.4

Things we found:
It seems mostly related to TCP sockets. 

With 
devServer: {
    host: "localhost"
  }

The devServer only creates a tcp4 socket. When the error happens, we saw tcp packets going to the destination ::1 port 3000 with the reset flag set.

I tested starting Nuxt dev server in dual-stack "yarn run dev --host ::", the issue is still there.

Launching Google Chrome with a custom hostname map works well.

open -a "Google Chrome" --args \               
  --host-resolver-rules="MAP localhost 127.0.0.1"

**Comment by flakerimi** [comment]
You need to bun lint and bun typecheck, code has problems.
